scrc/dataset_creation/judgment_dataset_creator.py

```diff
@@ -49,8 +57,48 @@ def plot_custom(self, df, split_folder, folder):
         self.plot_labels(df, split_folder, label_name='label')
 
 
+def create_multiple_datasets(court_names, exclude_courts):
+    not_created = []
+    created = []
+    for court_string in tqdm(court_names):
+        if court_string not in exclude_courts:
+            print(f"Creating dataset for {court_string}")
+            got_created = judgment_dataset_creator.create_dataset(court_string=court_string, sub_datasets=False, kaggle=False,
+                                                            save_reports=True)
+            if got_created:
+                # rename judgment_prediction folder to {court_string}
+                source = str(ROOT_DIR) + '/data/datasets/judgment_prediction'
+                destination = str(ROOT_DIR) + '/data/datasets/' + court_string
+                os.rename(source, destination)
+                created.append(court_string)
+            else:
+                print(f"Dataset for {court_string} could not be created")
+                not_created.append(court_string)
+            print("Empty courts: ", not_created)
+    print(f"{len(not_created)} courts not created: ", not_created)
+    print(f"{len(created)} courts created: ", created)
+
+
 if __name__ == '__main__':
     config = get_config()
 
-    judgment_dataset_creator = JudgmentDatasetCreator(config)
-    judgment_dataset_creator.create_dataset(sub_datasets=True, kaggle=False, save_reports=True)
+    # get names of all courts
+    court_strings = get_all_courts()
+
+    # taking all folder names from /data/datasets as a list to know which courts are already generated
+    courts_done = os.listdir(os.path.join(ROOT_DIR, "data", "datasets"))
+
+    # all courts that couldn't be created
+    courts_error = get_error_courts()
+
+    # all courts that can be generated but with some issues
+    courts_issues = get_issue_courts()
+
+    judgment_dataset_creator = JudgmentDatasetCreator(config, debug=False)
+
+    create_multiple_datasets(court_strings, exclude_courts=courts_done + courts_error + courts_issues)
+    # 76/183 not created
+    # 107/183 created - 2 without reports, 6 without file_numbers
```

Comment:
Ideally, in the main call we only want to create the Creator and invoke one function.
In the pretraining_dataset_creator, I have similar problems as you have here. Let's discuss this at some point.